src/ZkFold/Base/Protocol/ARK/Plonk/Relation.hs

```diff
@@ -0,0 +1,74 @@
+{-# LANGUAGE TypeOperators #-}
+
+module ZkFold.Base.Protocol.ARK.Plonk.Relation where
+
+import           Data.Map                                     (Map, elems, (!))
+import           GHC.IsList                                   (IsList (..))
+import           Numeric.Natural                              (Natural)
+import           Prelude                                      hiding (Num (..), drop, length, replicate, sum, take,
+                                                               (!!), (/), (^))
+
+import           ZkFold.Base.Algebra.Basic.Class
+import           ZkFold.Base.Algebra.Basic.Number
+import           ZkFold.Base.Algebra.Basic.Permutations       (Permutation, fromCycles, mkIndexPartition)
+import           ZkFold.Base.Algebra.Polynomials.Multivariate (evalMonomial, evalPolynomial, var)
+import           ZkFold.Base.Algebra.Polynomials.Univariate   (PolyVec, toPolyVec)
+import           ZkFold.Base.Data.Vector                      (Vector, fromVector)
+import           ZkFold.Base.Protocol.ARK.Plonk.Constraint    (PlonkConstraint (..), toPlonkConstraint)
+import           ZkFold.Prelude                               (replicate)
+import           ZkFold.Symbolic.Compiler
+
+type PlonkPolyDegree n = 2 ^ (Log2 n + 1)
```

Comment:
I think this is cool, but I don't understand where it could be used in type level.

Reply:
Yep, forgot to delete it. Since we do not know the circuit size at compile time, we cannot determine a proper Plonk polynomial degree at compile time.